### PR TITLE
all: implement mocktracer in terms of the new API

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This PR implements the mocktracer package in terms of the new tracing APIs. The goal here is to have as few changes as possible to the mocktracer API and therefore to the contrib tests.

This is not yet complete, but it is working for the `gin-gonic` contrib package. This is a good example of what I'm going for and the (hopefully) minimal changes that will be required for the other contrib packages.

These changes need to go along with https://github.com/DataDog/system-tests/pull/1935 which update the parametric test to use the new API. For now, that branch has been hard-coded in the CI for this PR, but that will be changed before merging.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: v2/contrib/urfave/negroni/go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/DataDog/dd-trace-go/v2/contrib/urfave/negroni
 
 go 1.19
 
+replace github.com/DataDog/dd-trace-go/v2 => ../../../../
+
```
Comment (darccio) [inline v2/contrib/urfave/negroni/go.mod:6]: @knusbaum PR looks good in general. About this replace directives, if you merge v2-dev, all contribs now have them too.

Comment (knusbaum) [reply v2/contrib/urfave/negroni/go.mod:6]: Thanks, that's very good to know.

File: docker-compose.yaml [hunk 2]
```diff
@@ -82,6 +82,19 @@ services:
...
+      http.port: 9204-9300
+      discovery.type: single-node
+      xpack.security.enabled: "false"
+      ES_JAVA_OPTS: "-Xms750m -Xmx750m" # https://github.com/10up/wp-local-docker/issues/6
+    ports:
+      - "9204:9204"
+    ulimits:
+      nofile:
+        soft: 65536
+        hard: 65536
```
Comment (darccio) [inline docker-compose.yaml:97]: Did some tests failed because this was missing?

**Review by darccio — APPROVED** [review]
LGTM

File: docker-compose.yaml (hunk 2 above)
Comment (knusbaum) [reply docker-compose.yaml:97]: Yes, the elasticsearch 8 tests were failing with `test.sh` without this.

**Comment by knusbaum** [comment]
Merged this into the other branch, with PR in #2408

**Pull request merged**